lptree.pas

```diff
@@ -3033,7 +3036,8 @@ function TLapeTree_InternalMethod_High.isConstant: Boolean;
     if (FParams.Count = 1) and (not isEmpty(FParams[0])) then
     begin
       ParamType := FParams[0].resType();
-      if (ParamType <> nil) and (not (ParamType is TLapeType_Type)) and (ParamType.BaseType = ltDynArray) then
+      if (ParamType <> nil) and (not (ParamType is TLapeType_Type)) and
+        (ParamType.BaseType in LapeStringTypes+[ltDynArray]) then
```

Comment:
This has been covered in IRC, where we concluded that shortstring's size isn't constant.